src/clj/ataru/tarjonta_service/tarjonta_service.clj

```diff
@@ -17,3 +18,139 @@
                                   {} haut)))
             {}
             (client/get-forms-in-use (if in-oph-organization? nil all-organization-oids)))))
+
+(defprotocol TarjontaService
+  (get-hakukohde [this hakukohde-oid])
+  (get-haku [this haku-oid]))
+
+(defrecord CachedTarjontaService []
+  component/Lifecycle
+  TarjontaService
+
+  (start [this] this)
+  (stop [this] this)
+
+  (get-hakukohde [this hakukohde-oid]
+    (.cache-get-or-fetch (:cache-service this) :hakukohde hakukohde-oid #(client/get-hakukohde hakukohde-oid)))
+
+  (get-haku [this haku-oid]
+    (.cache-get-or-fetch (:cache-service this) :haku haku-oid #(client/get-haku haku-oid))))
+
+(defn new-tarjonta-service
+  []
+  (->CachedTarjontaService))
```

Comment:
For browser-test mode, I think we should use the Fake-version, like in organization_service:

`(if (-> config :dev :fake-dependencies)
    (->FakeTarjontaService)
    (->CachedTarjontaService))
`